### ISSUE TITLE
PDS job can be restarted

### ISSUE THREAD

**Issue by de-jcup** [mercedes-benz/sechub#1465]
## Situation
A JVM with a running PDS job has been crashed. The operation was long running on a remote site (e.g. Checkmarx, see #1415 )
When we start just a new PDS job for the same setup (e..g codescan), the results on the remote site would be already available and we would waste time.

## Wanted
"As an PDS administrator I want to restart a PDS job"

## Solution
- introduce REST API
- logic: 
   - fail when JOB is in state DONE or FAILED
     (reason: we do not want side effects with SecHub jobs - here the PDSAdapter does inspect the PDS job states.
     When already failed or done Sechub job will no try to fetch data again!)
  - Introduce new PDS job state: `PREPARE_RESTART`
     - (old still running) PDS job  must inspect, if the job state has changed to this new state.
       If so, PDS for former process (if still alive) will 
       - Log the restart event
       - terminate the former process hard
       - kill the current workspace (like for failed or done state)
       - change the 
- After #1474 is implemented, the same cluster logic must be implemented for `PREPARE_RESTART` but set
   to state `READY_FOR_RESTART`
   - (new) PDS job: wait for PDS job to be in state `READY_FOR_RESTART`   
     (if accidently there was still any other process in the cluster doing the job, it will be 
     terminated)
   - after ins state `READY_FOR_RESTART` or a time out for wait (means JVM of old job is crashed) , change the PDS job state into READY_TO_START  
     (so will be executed again after some time)

- document: describe the scenario when a PDS JVM crashes and we have a long running operation on remote side
  e.g. on #1415

**Comment by de-jcup** [comment]
This is a duplicate of #1457 which is already implemented, so closing this one.
